core/state/dump.go

```diff
@@ -19,32 +19,68 @@ package state
 import (
 	"encoding/json"
 	"fmt"
+	"os"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/ethereum/go-ethereum/trie"
 )
 
+// DumpAccount represents an account in the state
 type DumpAccount struct {
 	Balance  string            `json:"balance"`
 	Nonce    uint64            `json:"nonce"`
 	Root     string            `json:"root"`
 	CodeHash string            `json:"codeHash"`
 	Code     string            `json:"code"`
 	Storage  map[string]string `json:"storage"`
+	Address  *string           `json:"address,omitempty"` // Address only present in iterative (line-by-line) mode
 }
 
+// Dump represents the full dump in a collected format, as one large map
 type Dump struct {
 	Root     string                 `json:"root"`
 	Accounts map[string]DumpAccount `json:"accounts"`
```

Comment:
Shouldn't we use `map[common.Address]DumpAccount` instead of `map[string]DumpAccount`?